### PR TITLE
use "-T linux" if $TERM is dumb or empty

### PR DESCRIPTION
**Pull request by Dudemanguy**
Sorry for sending this one to a mirror but gitea registration is closed.

It's possible for `$TERM` to be `dumb` on bootup which causes `tput colors` to fail and thus the rest of the script. Workaround this by simply assuming $TERM can be linux if it is `dumb`.

### CODE REVIEW

**Comment by thefallenrat** [comment]
Hmm, Has this been tested on users with `$TERM` set to empty? Also, I'm interested if we could follow the **openRC** way to set colors in terminal

https://github.com/OpenRC/openrc/blob/5427783fdf3d183ea4e63afc507c31d88f0d2c9c/etc/rc.in#L14

**Comment by Dudemanguy** [comment]
I don't think `$TERM` should be set manually here since I believe whatever init system you're using should do that at some point. At least when I've been testing this with s6, the `$TERM` variable gets set sometime after the runit-rc scripts get run hence why I need this patch. That being said, it definitely makes sense to include an empty `$TERM` in the check as well. `tput colors` expects a value for `$TERM` so an empty `$TERM` will also fail.

I guess the weakness here is that it assumes if your `$TERM` is empty or dumb, it will at some point get set to something "proper" later (i.e. `linux` or something else that has color support).

File: script/functions.in [hunk 1]
```diff
@@ -201,7 +201,11 @@ run_shutdown(){
 calc_columns
 
 # disable colors on broken terminals
-TERM_COLORS=$(tput colors 2>/dev/null)
+if [[ -z "$TERM" -o "$TERM" = "dumb" ]]; then
```
Comment (konimex) [inline script/functions.in]: Use `||` instead of `-o`.

**Review by thefallenrat — APPROVED** [review]

**Review by konimex — CHANGES_REQUESTED** [review]

File: script/functions.in (hunk 1 above)
Comment (thefallenrat) [reply script/functions.in]: Don't think you need || for that

Comment (konimex) [reply script/functions.in]: Rationale: https://github.com/koalaman/shellcheck/wiki/SC2110

On that note, I should probably rewrote the rc to be shellcheck compliant.

Comment (Dudemanguy) [reply script/functions.in]: Makes sense to me.

**Review by konimex — APPROVED** [review]

File: script/functions.in (hunk 1 above)
Comment (thefallenrat) [reply script/functions.in]: Then don't use `[[]]`, use `[]` instead

Comment (konimex) [reply script/functions.in]: If it's POSIX sh, then I'd be fine with `[`. But bash has `[[`, why don't we use it to our advantage?

Comment (konimex) [reply script/functions.in]: Also, see https://github.com/koalaman/shellcheck/wiki/SC2166

**Comment by konimex** [comment]
Merged manually. See https://github.com/artix-linux/runit-rc/commit/451fe75e69527cf656451517def18f0e07193675

**Pull request closed without merging**